### PR TITLE
Update govuk-components to v2

### PR DESCRIPTION
**Pull request by paulrobertlloyd**
### Context

A new version of [`govuk-components`](https://github.com/DFE-Digital/govuk-components) has been released. It contains a number of breaking changes, so this PR addresses any issues caused by moving to the new version.

### Changes proposed in this pull request

* Update Gemfile to point to v2.0.1
* Fix issues arising from breaking changes in the components used by this app
* The cookie banner component is now one component that takes multiple messages; updated the JavaScript (and related tests) to reflect this change. 

### Guidance to review

Might want to pay particular attention to the changes around the cookie banner component.

FYI @peteryates

### Checklist

- [x] Rebased master
- [x] Cleaned commit history
- [x] Tested by running locally

### CODE REVIEW

**Comment by paulrobertlloyd** [comment]
[Upgrade BAT services to use govuk-components v2](https://trello.com/c/Ah1LYNps/3860-upgrade-bat-services-to-use-govuk-components-v2)

**Comment by paulrobertlloyd** [comment]
This is first govuk-components v2 update PR that is ready to review, and probably the simplest of them!

**Review by duncanjbrown — APPROVED** [review]
Rebased, corrected a couple of banners where we included a `=` in the ERB tag for the heading, looked at all pages next to prod versions. Think this can go :+1:

File: app/views/results/index.html.erb [hunk 1]
```diff
@@ -43,6 +43,9 @@
             <% end %>
           <% end %>
 
+          <p class="govuk-body">
+            <%= govuk_link_to 'New search', root_path, no_visited_state: true %>
+          </p>
```
Comment (duncanjbrown) [inline app/views/results/index.html.erb:48]: @paulrobertlloyd I think this was removed in `master`?

Comment (duncanjbrown) [reply app/views/results/index.html.erb:48]: at  4356b82a8f65d321485aec4cc3d9cd5f1ddba

Comment (paulrobertlloyd) [reply app/views/results/index.html.erb:48]: It was. Thought I removed it in my latest rebase. Did I not?

**Pull request merged**